Cod3rsGrowth.Infra/Repositorios/RepositorioCarro.cs

```diff
@@ -75,11 +76,12 @@ public void Remover(int Id)
                  .Delete();
         }
 
-        private static IQueryable<Carro> FiltroParaBusca(ITable<Carro> carros, FiltroCarro carro)
+        private static IQueryable<Carro> FiltroParaBusca(ITable<Carro> carros, FiltroCarro? carro)
         {
             var query = carros.AsQueryable();
 
             if (carro != null)
```

Comment:
Inverte a lógica, seu código vai ficar mais limpo.
Ex:
if(carro is null) return query;

if(carro.Modelo is not null)
     query = query.Where(d => d.Modelo.Contains(carro.Modelo)); //verifica se isso é um tipo ou uma string

if...